src/basics.md

```diff
@@ -270,6 +271,40 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-verify-extract-email]: #ex-verify-extract-email
+<a name="ex-verify-extract-email"></a>
+## Verify and extract login from an email address
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Validates that an email address is formatted correctly, and extracts everything
+before the @ symbol.
+
+```rust,no_run
+extern crate regex;
+use regex::Regex;
+
+// A simple email validator and extractor. Validates that something has an @
+// symbol, followed by something that looks like a domain.
+// Returns what came before the @ symbol if everything looks valid.
+fn extract_login(input: &str) -> Option<String> {
```

Comment:
- Option<Cow<str>> return type would be nicer . 
- using lazy_static seams to be the common idiom when compiling the regex

https://github.com/brson/rust-cookbook/pull/248 is a fine example of both